packages/adapter-cloudflare-workers/index.js

```diff
@@ -1,152 +1,64 @@
 import { existsSync, readFileSync, writeFileSync } from 'node:fs';
 import { posix, dirname } from 'node:path';
-import { execSync } from 'node:child_process';
-import esbuild from 'esbuild';
 import toml from '@iarna/toml';
 import { fileURLToPath } from 'node:url';
 import { getPlatformProxy } from 'wrangler';
 
 /**
  * @typedef {{
  *   main: string;
- *   site: {
- *     bucket: string;
+ *   assets: {
+ *     directory: string;
+ *     binding: string;
  *   }
- *   compatibility_flags?: string[];
  * }} WranglerConfig
  */
 
-// list from https://developers.cloudflare.com/workers/runtime-apis/nodejs/
-const compatible_node_modules = [
-	'assert',
-	'async_hooks',
-	'buffer',
-	'crypto',
-	'diagnostics_channel',
-	'events',
-	'path',
-	'process',
-	'stream',
-	'string_decoder',
-	'util'
-];
-
 /** @type {import('./index.js').default} */
 export default function ({ config = 'wrangler.toml', platformProxy = {} } = {}) {
 	return {
 		name: '@sveltejs/adapter-cloudflare-workers',
 
-		async adapt(builder) {
-			const { main, site, compatibility_flags } = validate_config(builder, config);
-
+		adapt(builder) {
+			const { main, assets } = validate_config(builder, config);
 			const files = fileURLToPath(new URL('./files', import.meta.url).href);
-			const tmp = builder.getBuildDirectory('cloudflare-workers-tmp');
-
-			builder.rimraf(site.bucket);
-			builder.rimraf(dirname(main));
-
-			builder.log.info('Installing worker dependencies...');
-			builder.copy(`${files}/_package.json`, `${tmp}/package.json`);
-
-			// TODO would be cool if we could make this step unnecessary somehow
-			const stdout = execSync('npm install', { cwd: tmp });
-			builder.log.info(stdout.toString());
+			const outDir = dirname(main);
+			const relativePath = posix.relative(outDir, builder.getServerDirectory());
```

Comment:
```suggestion
			const out_dir = dirname(main);
			const relative_path = posix.relative(outDir, builder.getServerDirectory());
```
We use snake case for internal code.